app/src/components/adminConsole/AdminConsole.tsx

```diff
@@ -1,38 +1,32 @@
 import React, { useState, useEffect, useCallback } from "react";
 import SnappableManager from "./SnappableManager";
 import CurrentCup from "./currentCup/CurrentCup";
-import { getExistsUnpublished } from "../../firebase/cups/CupService";
+import { getCurrentCupIfExists } from "../../firebase/cups/CupService";
+import Cup from "../../types/Cup";
 
 const AdminConsole = () => {
-    const [isCup, setIsCup] = useState<Boolean>(false);
+    const [cup, setCup] = useState<Cup | undefined>(false);
 
-    useEffect(() => {
-        getExistsUnpublished()
-            .then((res: Boolean) => {
-                setIsCup(res);
-            })
-            .catch((e) => console.log(e));
-    }, [getExistsUnpublished, setIsCup]);
+    const updateCup = useCallback(() => {
+        (async () => {
+            try {
+                const res = await getCurrentCupIfExists();
+                setCup(res);
+            } catch (err) {
+                console.error(err);
+            }
+        })();
+    }, [setCup]);
```

Comment:
I don't think we need `useCallback`. This should be inside the body of `useEffect` instead.

Reply:
done - it's still defined in this way, and is called in useEffect